HIPs/incubation/xkey.md

```diff
@@ -0,0 +1,136 @@
+  secure arithmetic operations, while managing small subgroups and memory security.
+* Common primitives and ZK proofs like Schnorr Proof of Knowledge, unambiguous hashing,
+  secret sharing, and more.
+* Applied libraries like fast Paillier encryption, using CRT to speed up (de)encryption
+  and homomorphic operations, among others.
+
+The ultimate goal of this proposal is to make developing and maintaining TSS/MPC protocols 
+easier. We want to improve their reliability, readability, and security because we believe 
+these are the safest and most feature-rich ways to protect and move digital assets.
+
+# Status
+Pending proposal
+
+# Solution
+We developed following crates in Rust:
+
+* A framework called [`round-based`] that allows implementing MPC as a single function.
+  It abstracts away communication layer, and provides reusable tools for dealing with
+  MPC-specific types of communications.
+* Library for generic elliptic curve cryptography [`generic-ec`]. It allows writing code
+  generic over choice of curve, provides advanced tools like multiscalar multiplication.
+  It was designed with security in mind: it's resistent to small-group attacks, and it
+  provides things like `SecretScalar` which stores secrets on heap and zeroizes it when
+  it's not used anymore.
+  * [`generic-ec-zkp`] provides ZK proofs and tools based on `generic-ec` crate
+  * [`stark-curve`] exposes a pure Rust implementation of stark curve
+* [`slip-10`] is a crate for HD-wallet derivation that follows [slip10][slip10-spec]
+  standard (which is compatible with [bip32][bip32-spec])
+* [`udigest`] is a library for unambiguous hashing of structured data
+* [`fast-paillier`] contains fast implementation of paillier encryption that contains
+  optimizations such as using Chinese Remainder Theorem (CRT) when private key is known
+* And, finally, [`cggmp21`], an audited implementation of threshold ECDSA protocol based
+  on [CGGMP21 paper][cggmp21-paper].
+
+Currently in development:
+* An implementation of threshold Schnorr/EdDSA protocol following the
+  [FROST IETF draft][frost-draft].
+* Tools for running TSS in enclaves
+
+[`round-based`]: https://github.com/dfns/round-based
+[`generic-ec`]: https://github.com/dfns/generic-ec
+[`generic-ec-zkp`]: https://docs.rs/generic-ec-zkp
+[`stark-curve`]: https://github.com/dfns/stark-curve
+[`slip-10`]: https://github.com/dfns/slip-10
+[`udigest`]: https://github.com/dfns/udigest
+[`fast-paillier`]: https://github.com/dfns/fast-paillier
+[`cggmp21`]: https://github.com/dfns/cggmp21
+
+[cggmp21-paper]: https://eprint.iacr.org/2021/060 
+[frost-draft]: https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-frost-15
+
+[slip10-spec]: https://github.com/satoshilabs/slips/blob/master/slip-0010.md
+[bip32-spec]: https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki
+
+All our work listed above is currently licensed under dual MIT or Apache-2.0 licenses. 
+We're planning on relicensing the code to Apache-2.0 only.
```

Comment:
Do we need to relicense?

Reply:
Hart said it's a strong requirement and a good thing to do in any case, for sake of avoiding future problems related to the licensing and contributions